backend/src/api/workflows.ts

```diff
@@ -75,4 +74,72 @@ router.delete("/distribution/:id", async (req, res) => {
   res.status(200).end();
 });
 
+/**
+ * Create a new task workflow.
+ */
+ router.post("/task", async (req, res) => {
+  const workflow: TaskWorkflowModel = req.body;
+
+  const savedWorkflow = await dalWorkflow.create(
+    WorkflowType.TASK,
+    workflow
+  );
+  res.status(200).json(savedWorkflow);
+});
+
+/**
+ * Update an existing task workflow.
+ */
+router.post("/task/:id", async (req, res) => {
+  const id = req.params.id;
+  const { name, active, source, destinations, prompt, 
+    requiredActions, optionalActions, assignedGroups, postsPerGroup } = req.body;
+
+  const workflow: Partial<TaskWorkflowModel> = Object.assign(
+    {},
+    name === null ? null : { name },
+    active === null ? null : { active },
+    source === null ? null : { source },
+    destinations === null ? null : { destinations },
+    prompt === null ? null : { prompt },
+    requiredActions === null ? null : { requiredActions },
+    optionalActions === null ? null : { optionalActions },
```

Comment:
I think we can remove the `optionalActions` field for now. I know it's in the mockups but we can leave it out for now.

Reply:
@zarifmahmud Still need to remove this.

Reply:
Ah sorry, didn't see that it was also in workflow.ts